src/lib.rs

```diff
@@ -391,10 +391,66 @@ impl Event {
     /// Returns `Some(true)` if the connection has failed, `Some(false)` if the connection has not failed,
     /// or `None` if the platform does not support detecting this condition.
     #[inline]
+    #[deprecated(
+        since = "3.4.0",
+        note = "use `is_err` in combination of is_hub instead, see documentation for `is_err`"
+    )]
     pub fn is_connect_failed(&self) -> Option<bool> {
         self.extra.is_connect_failed()
     }
 
+    /// Tells if this event is the result of a connection failure.
+    ///
+    /// This function checks if an error exist,particularlly useful in detecting if TCP connection failed. It corresponds to the `EPOLLERR` event in Linux
+    /// and `CONNECT_FAILED` event in Windows IOCP.
+    ///
+    /// note: in epoll, a tcp connection failure is indicated by EPOLLERR + EPOLLHUP, though just EPOLLERR is enough to indicate a connection failure.
+    /// EPOLLHUP may happen when we haven't event called `connect` on the socket, but it is still a valid event to check for.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    ///   fn main() -> std::io::Result<()> {
+    ///       use std::net;
+    ///       use std::time::Duration;
+    ///   
+    ///       use polling::Event;
+    ///       use socket2::Type;
+    ///       let poller = polling::Poller::new()?;
+    ///       let bad_socket = socket2::Socket::new(socket2::Domain::IPV4, Type::STREAM, None)?;
+    ///       let addr = net::SocketAddr::new("127.0.0.1".parse().unwrap(), 12345);
+    ///       bad_socket.set_nonblocking(true)?;
+    ///       let mut events = polling::Events::new();
+    ///   
+    ///       let _ = bad_socket.connect(&addr.into()).unwrap_err();
+    ///   
+    ///       unsafe {
+    ///           poller.add(&bad_socket, Event::writable(0))?;
+    ///       }
+    ///   
+    ///       poller.wait(&mut events, Some(Duration::from_secs(3)))?;
+    ///   
+    ///       let event = events.iter().next().expect("no event");
+    ///       let is_err = match event.is_err() {
+    ///           Some(is_err) => is_err,
+    ///           None => {
+    ///               println!("not supported in this platform");
+    ///               return Ok(());
+    ///           },
+    ///       };
+    ///       assert!(is_err);
+    ///       println!("bad socket is now in error state");
+    ///   
+    ///       Ok(())
+    ///   }
+    /// ```  
+    /// Returns `Some(true)` if the connection has failed, `Some(false)` if there is an error,
```

Comment:
```suggestion
    /// ```  
    ///
    /// Returns `Some(true)` if the connection has failed, `Some(false)` if there is an error,
```